tt_metal/impl/buffers/buffer.cpp

```diff
@@ -248,7 +248,7 @@ Buffer &Buffer::operator=(Buffer &&other) {
 
 void Buffer::allocate() {
     TT_ASSERT(this->device_ != nullptr);
-    // L1 buffers are allocated top down!
+    // L1 and Trace buffers (which live in DRAM) are allocated top down!
```

Comment:
the `bottom_up` bool below evaluates to true if buffer is in DRAM so the comment is a bit confusing? do Trace buffers need to be allocated top down within the trace buffer region? right now it seems that they are allocated bottom up but at the top of a DRAM channel.

Reply:
ah okay I see. Maybe we need to differentiate between MemoryRegion and BufferType (we also have BufferLayout interleaved / sharded)